controllers/istio/istio_controller.go

```diff
@@ -160,7 +161,25 @@ func (r *IstioReconciler) getActiveRevision(ctx context.Context, istio *v1alpha1
 }
 
 func (r *IstioReconciler) getActiveRevisionKey(istio *v1alpha1.Istio) types.NamespacedName {
-	return types.NamespacedName{Name: istio.Name}
+	return types.NamespacedName{
+		Name: r.getActiveRevisionName(istio),
+	}
+}
+
+func (r *IstioReconciler) getActiveRevisionName(istio *v1alpha1.Istio) string {
+	var strategy v1alpha1.UpdateStrategyType
```

Comment:
nit: maybe just personal preference, but I'd rather use `strategy :=  v1alpha1.UpdateStrategyTypeInPlace` here. That would make the switch below simpler of even unnecessary...

Reply:
Unfortunately, this wouldn't change anything. The switch statement would still have to have a `default` case with a `fallthrough`, or there would have to be an additional return statement.